spyder/plugins/framesexplorer/widgets/main_widget.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Frames Explorer Main Plugin Widget.
+"""
+
+# Third party imports
+from qtpy.QtCore import Signal, Slot
+
+# Local imports
+from spyder.api.translations import get_translation
+from spyder.config.manager import CONF
+from spyder.config.gui import get_color_scheme
+from spyder.plugins.framesexplorer.widgets.framesbrowser import (
+    FramesBrowser)
+from spyder.api.shellconnect.main_widget import ShellConnectMainWidget
+
+# Localization
+_ = get_translation('spyder')
+
+
+# =============================================================================
+# ---- Constants
+# =============================================================================
+class FramesExplorerWidgetActions:
+    # Triggers
+    Search = 'search'
+    Refresh = 'refresh'
+    PostMortemDebug = 'pmdebug'
+
+    # Toggles
+    ToggleExcludeInternal = 'toggle_exclude_internal_action'
+    ToggleCaptureLocals = 'toggle_capture_locals_action'
+    ToggleLocalsOnClick = 'toggle_show_locals_on_click_action'
+
+
+class FramesExplorerWidgetOptionsMenuSections:
+    Display = 'excludes_section'
+    Highlight = 'highlight_section'
+
+
+class FramesExplorerWidgetMainToolBarSections:
+    Main = 'main_section'
+
+
+class FramesExplorerWidgetMenus:
+    EmptyContextMenu = 'empty'
+    PopulatedContextMenu = 'populated'
+
+
+class FramesExplorerContextMenuSections:
+    Locals = 'locals_section'
+
+
+class FramesExplorerContextMenuActions:
+    ViewLocalsAction = 'view_locals_action'
+
+
+# =============================================================================
+# ---- Widgets
+# =============================================================================
+
+
+class FramesExplorerWidget(ShellConnectMainWidget):
+
+    # PluginMainWidget class constants
+    ENABLE_SPINNER = True
+
+    # Signals
+    edit_goto = Signal((str, int, str), (str, int, str, bool))
+
+    def __init__(self, name=None, plugin=None, parent=None):
+        super().__init__(name, plugin, parent)
+
+        # Widgets
+        self.context_menu = None
+        self.empty_context_menu = None
+
+    def postmortem(self):
+        """Ask for post mortem debug."""
+        self.current_widget().shellwidget.execute("%debug")
+
+    def capture_frames(self):
```

Comment:
Please move these two methods to the Public API section below.

Reply:
You missed this one too. The idea is to move `postmortem` and `capture_frames` below the `view_item_locals` method because they are part of the public API of this widget.